server/pulp/server/webservices/views/consumers.py

```diff
@@ -0,0 +1,1082 @@
+        :type content_type: str
+
+        :return: Response representing the updated profile
+        :rtype: django.http.HttpResponse
+        """
+
+        body = request.body_as_json
+        profile = body.get('profile')
+
+        manager = factory.consumer_profile_manager()
+        consumer = manager.update(consumer_id, content_type, profile)
+
+        add_link_profile(consumer)
+
+        return generate_json_response_with_pulp_encoder(consumer)
+
+    @auth_required(authorization.DELETE)
+    def delete(self, request, consumer_id, content_type):
+        """
+        Delete an association between the specified
+        consumer and profile.  Designed to be idempotent.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: A consumer ID.
+        :type consumer_id: str
+        :param content_type: The content type ID.
+        :type content_type: str
+
+        :return: An empty response
+        :rtype: django.http.HttpResponse
+        """
+
+        manager = factory.consumer_profile_manager()
+        response = manager.delete(consumer_id, content_type)
+        return generate_json_response(response)
+
+
+class ConsumerContentApplicabilityView(View):
+    """
+    View for query content applicability.
+    """
+    @auth_required(authorization.READ)
+    @json_body_required
+    def post(self, request):
+        """
+        Query content applicability for a given consumer criteria query.
+
+        body {criteria: <object>,
+              content_types: <array>[optional]}
+
+        This method returns a JSON document containing an array of objects that each have two
+        keys: 'consumers', and 'applicability'. 'consumers' will index an array of consumer_ids,
+        for consumers that have the same repository bindings and profiles. 'applicability' will
+        index an object that will have keys for each content type that is applicable, and the
+        content type ids will index the applicability data for those content types. For example,
+
+        [{'consumers': ['consumer_1', 'consumer_2'],
+          'applicability': {'content_type_1': ['unit_1', 'unit_3']}},
+         {'consumers': ['consumer_2', 'consumer_3'],
+          'applicability': {'content_type_1': ['unit_1', 'unit_2']}}]
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+
+        :return: Response containing applicability data matching the consumer criteria query
+        :rtype:  jango.http.HttpResponse
+        """
+
+        # Get the consumer_ids that match the consumer criteria query that the requestor queried
+        # with, and build a map from consumer_id to a dict with profiles and repo_ids for each
+        # consumer
+        try:
+            consumer_criteria = self._get_consumer_criteria(request)
+            content_types = self._get_content_types(request)
+        except InvalidValue, e:
```

Comment:
if i remove the try/except block the response body will be different http://fpaste.org/204692/77159541/
I am not sure why it was meant to be this way, i think it is more readable - so i decided to leave the old code without any changes.